tools/build_config/build.sh

```diff
@@ -85,6 +85,41 @@ function run_package() {
   echo_and_run "cpack -G TGZ"
 }
 
+function build_docs() {
+    # Update release numbers
+    build_id=$(sed -nr '/CPACK_PACKAGE_FILE_NAME/{s/.*"Horace-([^"]+)".*/\\1/p};' ./build/CPackConfig.cmake)
+    sed -i -r 's/release = .*/release = "'${build_id}'"/' ./documentation/user_docs/docs/conf.py
+    make -C ./documentation/user_docs/ html
+    make -C ./documentation/user_docs/ html
+
+    # Undo change to allow checkout
+    git checkout ./documentation/user_docs/docs/conf.py
+
+    # Remove dead links
+    sed -i -r '/\\[NULL\\]/d' ./documentation/user_docs/build/html/*html
+
+    # Compress for artifact
+    cd ./documentation/user_docs/build/html
+    tar -czf docs.tar.gz ./*
+    cd -
+    mv ./documentation/user_docs/build/html/docs.tar.gz .
+}
+
+function push_built_docs() {
+    git config --local user.name "PACE CI Build Agent"
+    git config --local user.email "pace.builder.stfc@gmail.com"
+    git remote set-url --push origin "https://pace-builder:"\${api_token## }"@github.com/pace-neutrons/Horace"
```

Comment:
should the URL and email be passed in rather than hard-coded?

The powershell script trims the token before the `@`, this explicitly has `## `; is this difference platform dependent?

Reply:
It's a lazy in-place trim

`${var## }` and `${var%% }` are bash commands for stripping whitespace,  `#` -> strip front, `%` strip back, when doubled repeat operation. Perhaps best explored with 
```bash
var='test.tar.gz'
echo ${var%.*}
echo ${var%%.*}
```
To get the basename.

Reply:
**N.B.** `.*` there is not RegEx match anything, but bash style globbing and a literal dot, anything equivalent to `\..*` in RegEx